### ISSUE TITLE
Duplicate names must be prohibited, even if there are no conflicts

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#21]
Consider this program:

```
[] > foo
  [] > a1
    42 > bar
  [] > a2
    33 > bar
```

Here, the first `bar` doesn't conflict with the second one, but readability of such a program suffers. Let's issue a `warning` in this case (we most probably need a new lint for this).

**Comment by Marat-Tim** [comment]
Let me take it

**Comment by yegor256** [comment]
@Marat-Tim sure, please, go ahead

**Comment by Marat-Tim** [comment]
And what to do in such a situation?
```
while
    x.as-number.lt 6 > [i] >>
    [i] >>
...
```
it turns into
```
<o line="17" name="a🌵1734" pos="12">
    <o base="∅" line="17" name="i" pos="32"/>
    <o base="$.^.x.as-number.lt" line="17" name="@" pos="23">
        <o base="Q.org.eolang.number" line="17" pos="27">
            <o base="Q.org.eolang.bytes">40-18-00-00-00-00-00-00</o>
        </o>
    </o>
</o>
<o line="18" name="a🌵1815" pos="12">
    <o base="∅" line="18" name="i" pos="13"/>
    <o base="Q.org.eolang.seq" line="19" name="@" pos="14">
```
and there is a duplication of the name, but as if in this context it's not a bad thing

I had an idea that you can ignore objects named using `>>` syntax, but I don't understand how to understand that the name was given in this way

**Comment by Marat-Tim** [comment]
`i` use in lines 17 and 18(if you look at xmir)

**Comment by maxonfjvipon** [comment]
@Marat-Tim yes, but they used in the different scopes of abstract objects. It's the same as you use variables with the same names, but it different functions in java:
```java
void first() {
  int x;
}
void second() {
  int x;
}
```

**Comment by maxonfjvipon** [comment]
@Marat-Tim yes, you're right about `i`, but names under `>>` are unique because they contain line and position in themselves

**Comment by Marat-Tim** [comment]
It is clear that I can distinguish where which object is, but do I need to return a warning that the name attribute of the objects is the same?

**Comment by maxonfjvipon** [comment]
@Marat-Tim you need to "return" a warning in you've found the 2+ objects with the same names, that's it)

**Comment by Marat-Tim** [comment]
okay, it's just that this is an example from a file with the "canonical" eo code and it's strange that it may contain warnings

**Comment by h1alexbel** [comment]
@yegor256 the lint `duplicate-names-in-diff-context` was implemented, can we close this one?

**Comment by yegor256** [comment]
@h1alexbel thanks!
